jcl/src/openj9.jvm/share/classes/com/ibm/jvm/CRIUSupport.java

```diff
@@ -23,17 +22,54 @@
  *******************************************************************************/
 package com.ibm.jvm;
 
+import java.nio.charset.StandardCharsets;
+import java.nio.file.Path;
+
 /**
  * CRIU Support API
  */
 public final class CRIUSupport {
 	
-	private static final boolean criuSupportEnabled = isCRIUSupportEnabledImpl();
+	/**
+	 * CRIU operation return code
+	 */
+	public enum CRIUResult {
+		/**
+		 * The operation was successful.
+		 */
+		SUCCESS,
+		/**
+		 * The requested operation is unsupported.
+		 */
+		UNSUPPORTED_OPERATION,
+		/**
+		 * The arguments provided for the operation are invalid.
+		 */
+		INVALID_ARGUMENTS,
```

Comment:
When will we raise `INVALID_ARGUMENTS` result code ? i.e. how would this be checked ?

Reply:
This occurs if the checkpoint data directory is invalid. (ie. we cant open it, or its null)